### PR TITLE
feat: categorical colour states added with tests.

### PR DESCRIPTION
**Pull request by thmasjs**
# Changes

*Describe the changes made, including any changes to tests, dependencies or docs. Link to any relevant issues below.*

- Closes #XXXX *(replace XXXX with GitHub issue number)*
- ...

# Tasks

- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Has documentation (including user guide) been updated
- [ ] Have new dependencies been added (or changed) in relevant `requirements.txt`
- [ ] Code linting, tests and other workflows pass

### CODE REVIEW

File: packages/vis-core/src/utils/map.js [hunk 1]
```diff
@@ -40,6 +40,101 @@ export const getMetricDefinition = (
   return null;
 };
 
+export const buildCategoricalLegendKey = ({ fieldName, value }) => {
```
Comment (brereh) [inline packages/vis-core/src/utils/map.js:51]: This function requires a descriptive docstring

Comment (brereh) [reply packages/vis-core/src/utils/map.js:51]: as do the others

**Review by brereh — CHANGES_REQUESTED** [review]
Functionality works well. We just need to have descriptive docstrings for the functions. As the functionality has a large footprint, it'd be useful to include some common text in all the docstrings -- e.g. "relates to categorical legend cache functionality" so that we can tie them together. Otherwise it's difficult to trace the logic chain.

**Review by brereh — APPROVED** [review]
Thanks for the additional docstrings, happy with this now.
